### PR TITLE
Disable TLS in Linux wheels

### LINKED ISSUE #156: BUG: cannot allocate memory in static TLS block   when installing through pip
The installation wia conda froge was getting stuck in the "Solving environment" part so I tried to install with pip, given that the package is available on Pypi. `pip install glum` runs in seconds, but then I am unable to import stuff from it, with the following error:

```python
In [1]: from glum import GeneralizedLinearRegressor
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-0284693fe484> in <module>
----> 1 from glum import GeneralizedLinearRegressor

~/anaconda3/lib/python3.9/site-packages/glum/__init__.py in <module>
      1 import pkg_resources
      2 
----> 3 from ._distribution import TweedieDistribution
      4 from ._glm import GeneralizedLinearRegressor
      5 from ._glm_cv import GeneralizedLinearRegressorCV

~/anaconda3/lib/python3.9/site-packages/glum/_distribution.py in <module>
      6 import numpy as np
      7 from scipy import sparse, special
----> 8 from tabmat import MatrixBase, StandardizedMatrix
      9 
     10 from ._functions import (

~/anaconda3/lib/python3.9/site-packages/tabmat/__init__.py in <module>
----> 1 from .categorical_matrix import CategoricalMatrix
      2 from .constructor import from_csc, from_pandas
      3 from .dense_matrix import DenseMatrix
      4 from .matrix_base import MatrixBase
      5 from .sparse_matrix import SparseMatrix

~/anaconda3/lib/python3.9/site-packages/tabmat/categorical_matrix.py in <module>
    171 from .ext.split import sandwich_cat_cat, sandwich_cat_dense
    172 from .matrix_base import MatrixBase
--> 173 from .sparse_matrix import SparseMatrix
    174 from .util import (
    175     check_matvec_out_shape,

~/anaconda3/lib/python3.9/site-packages/tabmat/sparse_matrix.py in <module>
      4 from scipy import sparse as sps
      5 
----> 6 from .ext.sparse import (
      7     csc_rmatvec,
      8     csc_rmatvec_unrestricted,

ImportError: /home/mathurin/anaconda3/lib/python3.9/site-packages/tabmat/ext/../../tabmat.libs/libjemalloclocal-691a3dac.so.2: cannot allocate memory in static TLS block
```

googling did not help, is there a way to make the pip-installed version work ?

### PR DESCRIPTION
**Pull request by xhochy**
Fixes #156

Checklist
* [ ] Added a `CHANGELOG.rst` entry

### CODE REVIEW

**Comment by xhochy** [comment]
Based on the flags in https://github.com/conda-forge/jemalloc-feedstock/blob/4b37abb1cb4fa9b4fd117c8bd60d4ce94fef4742/recipe/build-jemalloc.sh#L21-L22

File: pyproject.toml [hunk 1]
```diff
@@ -38,7 +38,7 @@ before-all = [
   "cd ~/",
   "git clone --branch 5.2.1 https://github.com/jemalloc/jemalloc.git",
   "cd jemalloc",
-  "./autogen.sh --disable-cxx --with-jemalloc-prefix=local --with-install-suffix=local",
+  "./autogen.sh --disable-cxx --with-jemalloc-prefix=local --with-install-suffix=local --disable-tls --disable-initial-exec-tls",
```
Comment (MarcAntoineSchmidtQC) [inline pyproject.toml:41]: This will disable tls for macos, not linux. Line 72 is where the same setup is declared for linux.

**Comment by MarcAntoineSchmidtQC** [comment]
@xhochy, should we make sure that all the flags are the same? In particular, I see  `--disable-static` for all platforms and `--with-lg-page=14` for macos. These flags are not used when building the wheels.

**Review by MarcAntoineSchmidtQC — APPROVED** [review]
Thanks Uwe!

**Pull request merged**